app/models/pieces/bishop.rb

```diff
@@ -5,4 +5,8 @@ def set_defaults
     self.moved = false
     self.icon = (self.color == 1) ? "&#9821;" : "&#9815;"
   end
+
+  def valid_move?(x,y)
+  	return false unless ((x_position - x) / (y_position - y)) == 1 or -1
```

Comment:
Maybe a good use for a private method as discussed in the call:

return false unless valid_diagonal_move?(x,y) and putting the slope logic there